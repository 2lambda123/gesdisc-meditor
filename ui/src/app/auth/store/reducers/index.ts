import { createSelector, createFeatureSelector } from '@ngrx/store';
import * as fromAuth from './auth';

export interface AuthState {
  status: fromAuth.State;
}


export const reducers = {
  status: fromAuth.reducer
};

export const selectAuthState = createFeatureSelector<AuthState>('auth');

export const selectAuthStatusState = createSelector(
  selectAuthState,
  (state: AuthState) => state.status
);

export const getLoggedIn = createSelector(
  selectAuthStatusState,
  fromAuth.getLoggedIn
);

export const getUser = createSelector(
  selectAuthStatusState, 
  fromAuth.getUser
);