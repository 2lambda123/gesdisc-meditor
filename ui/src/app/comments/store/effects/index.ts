import { CommentsEffects } from './comments.effects';

export const effects: any[] = [CommentsEffects];

export * from './comments.effects';