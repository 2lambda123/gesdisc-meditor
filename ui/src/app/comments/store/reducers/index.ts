import {
	createFeatureSelector,
	ActionReducerMap,
} from '@ngrx/store';
import * as fromComments from './comments.reducer';

export interface CommentsState {
	comments: fromComments.State;
}

export const reducers: ActionReducerMap<CommentsState> = {
	comments: fromComments.reducer
};

export const getCommentsState = createFeatureSelector<CommentsState>('comments');



