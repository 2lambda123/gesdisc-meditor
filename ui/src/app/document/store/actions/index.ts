export * from './history.actions';
export * from './document.actions';