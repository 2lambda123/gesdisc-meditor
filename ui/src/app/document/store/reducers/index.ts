import {
	createFeatureSelector,
	ActionReducerMap,
} from '@ngrx/store';
import * as fromHistory from './history.reducer';
import * as fromDocument from './document.reducer';

export interface DocumentDataState {
	history: fromHistory.State;
	document: fromDocument.State;
}

export const reducers: ActionReducerMap<DocumentDataState> = {
	history: fromHistory.reducer,
	document: fromDocument.reducer
};

export const getDocumentDataState = createFeatureSelector<DocumentDataState>('documentData');


