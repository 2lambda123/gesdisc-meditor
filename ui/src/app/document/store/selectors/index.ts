export * from './document.selectors';
export * from './history.selectors';