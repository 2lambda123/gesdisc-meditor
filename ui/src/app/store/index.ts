export { DocumentState } from './document/document.state';
export { ModelState } from './model/model.state';
export { NotificationState } from './notification/notification.state';
export { AuthState } from './auth/auth.state';
export { RouterState } from './router/router.state';
