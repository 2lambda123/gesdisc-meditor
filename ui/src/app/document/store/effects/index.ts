import { HistoryEffects } from './history.effects';
import { DocumentEffects } from './document.effects';

export const effects: any[] = [HistoryEffects, DocumentEffects];

export * from './history.effects';
export * from './document.effects';