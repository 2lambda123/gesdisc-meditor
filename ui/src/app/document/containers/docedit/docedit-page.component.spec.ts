import { ComponentFixture, TestBed  } from '@angular/core/testing';
import { NgxsModule, Store, } from '@ngxs/store';
import { DocEditPageComponent } from './docedit-page.component';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { of } from 'rxjs';

describe('Document Edit Page', () => {
	let fixture: ComponentFixture<DocEditPageComponent>;
	let store: Store;
	let component: DocEditPageComponent;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [
				NoopAnimationsModule,
        NgxsModule.forRoot()
			],
			declarations: [
				DocEditPageComponent
			],
			schemas: [ NO_ERRORS_SCHEMA ]
		});

		fixture = TestBed.createComponent(DocEditPageComponent);
		component = fixture.componentInstance;

    Object.defineProperty(component, 'userPrivileges$', { writable: true });
    Object.defineProperty(component, 'document$', { writable: true });
    component.userPrivileges$ = of(['edit']);
    component.document$ = of({'x-meditor': {'targetStates': ['Draft', 'Under Review']}});
		fixture.detectChanges();
    
		store = TestBed.get(Store);
	});

	it('should compile', () => {
		expect(fixture).toMatchSnapshot();
	});

});
