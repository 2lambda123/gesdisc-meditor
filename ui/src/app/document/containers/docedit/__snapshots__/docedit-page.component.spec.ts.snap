// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document Edit Page should compile 1`] = `
<med-docedit-page
  __commentsCount$__selector={[Function Observable]}
  __edges$__selector={[Function Observable]}
  __history$__selector={[Function Observable]}
  __model$__selector={[Function Observable]}
  __versionsCount$__selector={[Function Observable]}
  __workflow$__selector={[Function Observable]}
  dirty="false"
  history="undefined"
  readonlydoc={[Function Boolean]}
  sidenav={[Function ElementRef]}
  store={[Function Store]}
  versionFilter={[Function Date]}
>
  <div
    class="doc-edit-container"
    fxlayout="row"
  >
    <mat-card
      class="mat-elevation-z0"
      fxflex="100"
    >
      <mat-drawer-container
        class="comment-sidenav-container"
      >
        <mat-drawer
          class="comment-sidenav"
        >
          
          
        </mat-drawer>
        <mat-drawer-content
          class="editor-form"
        >
          <med-doc-breadcrumbs />
          <mat-card-title>
            <i
              class="icon-badge icon-badge-sm fa "
            />
              
          </mat-card-title>
          <mat-card-subtitle>
              
          </mat-card-subtitle>
          <mat-card-content>
            <div
              class="button-panel"
            >
              <button
                mat-mini-fab=""
                matbadgecolor="warn"
              >
                <mat-icon>
                  question_answer
                </mat-icon>
              </button>
              <button
                mat-mini-fab=""
                matbadgecolor="warn"
              >
                <mat-icon>
                  history
                </mat-icon>
              </button>
              <mat-chip-list
                class="state-chip"
              >
                <mat-chip>
                  
                </mat-chip>
              </mat-chip-list>
              <span
                class="version-info"
              >
                (edited by  on )
              </span>
            </div>
            <med-document-edit
              class=""
            />
            
            <button
              disabled=""
              mat-raised-button=""
            >
              Save 
            </button>
            
            <med-document-actions />
          </mat-card-content>
        </mat-drawer-content>
      </mat-drawer-container>
    </mat-card>
  </div>
</med-docedit-page>
`;
